### PR TITLE
🐛 [amp-story] Audio bug fixes

### LINKED ISSUE #29520: [amp-story] Error: Cannot play unregistered element.
Using <amp-audio> under <amp-story-page> is unplayable and throwing the following error:

```
 Error: Cannot play unregistered element.
```

### PR DESCRIPTION
**Pull request by Enriqe**
Closes #29520

* Using `<amp-audio>` under `<amp-story-page` was throwing an error and audio was not being played (we had this issue in #28404). This is due to a recurring mistake on our side (reintroduced in #29233) on waiting for the `<amp-audio>` to be laid out, but since it's not actually laid out (because it's `layout="nodisplay"`) the promise never resolves and the element wasn't being registered. I'll add tests unit tests to prevent this from happening in the future.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-page.js [hunk 1]
```diff
@@ -714,6 +714,12 @@ export class AmpStoryPage extends AMP.BaseElement {
       return new Promise((resolve) => {
         switch (mediaEl.tagName.toLowerCase()) {
           case 'amp-audio':
+            if (mediaEl.getAttribute('layout') === Layout.NODISPLAY) {
+              whenUpgradedToCustomElement(mediaEl)
+                .then((el) => el.signals().whenSignal(CommonSignals.BUILT))
+                .then(resolve, resolve);
+              break;
+            }
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js:722]: It looks like amp-video supports `NODISPLAY` too... Could you move this block so we support both audio and video?

File: extensions/amp-story/1.0/amp-story-page.js [hunk 2]
```diff
@@ -1067,6 +1073,7 @@ export class AmpStoryPage extends AMP.BaseElement {
    * @return {!Promise} Promise that resolves after the callbacks are called.
    */
   unmuteAllMedia() {
+    this.playAudioElementFromTimestamp_ = Date.now();
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js:1076]: Do you need this? It should already be called when calling play()

**Comment by Enriqe** [comment]
Cleaned up this PR to solely focus on the `nolayout` issue, will send separate ones for the other issues.

File: extensions/amp-story/1.0/amp-story-page.js (hunk 1 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-page.js:723]: I think doing something like `const signal = mediaEl.getAttribute('layout') === Layout.NODISPLAY ? BUILT : LOAD_END;` so you don't have to repeat this block might be easier, but it LGTM :))

**Pull request merged**